plugins/module_utils/utils.py

```diff
@@ -96,10 +96,11 @@ def recursive_replace(data, path, keys, new_value):
         if len(keys) == 1:
             # Update the existing configuration
             if new_value is not None and data.get(key) != new_value:
+                operation = "replace" if data.get(key) is not None else "add"
```

Comment:
Did we test all the modules utilizing this helper function just to make sure that nothing is breaking?

Reply:
I tested the change using the latest NDO modules, and all tests passed without any issues.